runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,758 @@
+
+Examples:
+>lua
+    -- Normal mode mapping for Vim command
+    vim.keymap.set('n', '<Leader>ex1', '<cmd>echo "Example 1"<cr>')
+    -- Normal and Commandline mode mapping for Vim command
+    vim.keymap.set({'n', 'c'}, '<Leader>ex2', '<cmd>echo "Example 2"<cr>')
+    -- Normal mode mapping for Lua function
+    vim.keymap.set('n', '<Leader>ex3', vim.treesitter.start)
+    -- Normal mode mapping for Lua function with arguments
+    vim.keymap.set('n', '<Leader>ex4', function() print('Example 4') end)
+<
+You can map functions from Lua modules via
+>lua
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action)
+<
+Note that this loads the plugin at the time the mapping is defined. If you
+want to defer the loading to the time when the mapping is executed (as for
+|autoload| functions), wrap it in `function() end`:
+>lua
+    vim.keymap.set('n', '<Leader>pl2', function() require('plugin').action() end)
+<
+The fourth, optional, argument is a table with keys that modify the behavior
+of the mapping such as those from `:map-arguments`. The following are the most
+useful options:
+• `buffer`: If given, only set the mapping for the buffer with the specified
+  number; `0` or `true` means the current buffer. >lua
+    -- set mapping for the current buffer
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action, { buffer = true })
+    -- set mapping for the buffer number 4
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action, { buffer = 4 })
+<
+• `silent`: If set to `true`, suppress output such as error messages. >lua
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action, { silent = true })
+<
+• `expr`: If set to `true`, do not execute the {rhs} but use the return value
+  as input. Special |keycodes| are converted automatically. For example, the following
+  mapping replaces <down> with <c-n> in the popupmenu only: >lua
+    vim.keymap.set('c', '<down>', function()
+      if vim.fn.pumvisible() == 1 then return '<c-n>' end
+      return '<down>'
+    end, { expr = true })
+<
+• `desc`: A string that is shown when listing mappings with, e.g., |:map|.
+  This is useful since Lua functions as {rhs} are otherwise only listed as
+  `Lua: <number> <source file>:<line>`. Plugins should therefore always use this
+  for mappings they create. >lua
+    vim.keymap.set('n', '<Leader>pl1', require('plugin').action,
+      { desc = 'Execute action from plugin' })
+<
+• `remap`: By default, all mappings are nonrecursive by default (i.e.,
+  |vim.keymap.set()| behaves like |:noremap|). If the {rhs} is itself a mapping
+  that should be executed, set `remap = true`: >lua
+    vim.keymap.set('n', '<Leader>ex1', '<cmd>echo "Example 1"<cr>')
+    -- add a shorter mapping
+    vim.keymap.set('n', 'e', '<Leader>ex1', { remap = true })
+<
+  The exception is if the {rhs} is a special |<Plug>| mapping, for which
+  `remap = true` by default: >lua
+    vim.keymap.set('n', '[%', '<Plug>(MatchitNormalMultiBackward)')
```

Comment:
```suggestion
```

Reply:
Please explain.

Reply:
There is no such exception

Reply:
`<Plug>` gets expanded even with noremap

Reply:
There is no such exception in the Lua API

Reply:
This is not a discussion of _differences_; it's purely an explanation of how to use the Lua API with as little background as possible. But I can rephrase thist.

Reply:
OK, that makes sense. (It's a bit niche to cover for the purpose of this guide, but it shouldn't be misleading.)

Reply:
Actually one second, I need to confirm this

Reply:
Just like `<SNR>` in `<script>` mappings